### PR TITLE
perf: read calculated cost from observations table directly if available

### PR DESCRIPTION
**Pull request by hassiebp**
Read calculated cost from observations table directly if available

### CODE REVIEW

File: web/src/features/dashboard/components/UserChart.tsx [hunk 1]
```diff
@@ -34,7 +36,9 @@ export const UserChart = ({
   const user = api.dashboard.chart.useQuery(
     {
       projectId,
-      from: "traces_observationsview",
+      from: env.NEXT_PUBLIC_HAS_BACKFILLED_OBSERVATION_COST
```
Comment (maxdeichmann) [inline web/src/features/dashboard/components/UserChart.tsx:39]: Can you add it to the env.production file?
Maybe we should also rename this to NEXT_PUBLIC_LANGFUSE_3_MIGRATION_DONE or similar

Comment (marcklingen) [reply web/src/features/dashboard/components/UserChart.tsx]: fyi, we should try to not use NEXT_PUBLIC envs as they get flattened at build-time, thus, they are always null or static for our prebuilt docker images. Can we do this via a regular env that gets used server-side or NEXT_PUBLIC_CLOUD_ENV which only gets used in our cloud deployment and is passed at build-time in or cloud build pipeline?

File: packages/shared/prisma/migrations/20240701103900_observations_view_read_from_calculated/migration.sql [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    m.start_date AS "model_start_date",
+    m.input_price,
+    m.output_price,
+    m.total_price,
+    m.tokenizer_config AS "tokenizer_config",
+    CASE 
+        WHEN o.calculated_input_cost IS NULL AND o.input_cost IS NULL AND o.output_cost IS NULL AND o.total_cost IS NULL THEN
+            o.prompt_tokens::decimal * m.input_price
+        ELSE
+            COALESCE(o.calculated_input_cost, o.input_cost)
```
Comment (maxdeichmann) [inline packages/shared/prisma/migrations/20240701103900_observations_view_read_from_calculated/migration.sql:41]: Dont we want to return `o.input_cost` if it is not null before returning `o.calculated_input_cost`?

Comment (maxdeichmann) [reply packages/shared/prisma/migrations/20240701103900_observations_view_read_from_calculated/migration.sql:41]: Same for the output_cost.

File: web/src/features/dashboard/components/UserChart.tsx (hunk 1 above)
Comment (maxdeichmann) [reply web/src/features/dashboard/components/UserChart.tsx]: As discussed, lets take NEXT_PUBLIC_LANGFUSE_CLOUD_REGION here.

**Pull request merged**